### PR TITLE
Adds GetMessagesStatus method

### PR DESCRIPTION
**Pull request by thiagodeev**
https://github.com/NethermindEth/starknet.go/issues/623
https://github.com/NethermindEth/starknet.go/issues/624

This PR aims to implement some RPC v0.8.0 updates to starknet.go. The ones addressed by this PR are:

Methods
starknet_getMessagesStatus

Note: As the RPC v0.8.0 has not yet been released, we can't fully test these changes as the nodes haven't implemented them yet. They will be updated before merging into main

### CODE REVIEW

File: rpc/transaction.go [hunk 1]
```diff
@@ -71,3 +71,20 @@ func (provider *Provider) GetTransactionStatus(ctx context.Context, transactionH
...
+
+// Given an l1 tx hash, returns the associated l1_handler tx hashes and statuses for all L1 -> L2 messages sent by the l1 transaction, ordered by the l1 tx sending order
+//
+// Parameters:
+// - ctx: the context.Context object for cancellation and timeouts.
+// - transactionHash: The hash of the L1 transaction that sent L1->L2 messages as a felt
+// Returns:
+// - [] MessageStatusResp: An array containing the status of the messages sent by the l1 transaction
+// - error, if one arose.
+func (provider *Provider) GetMessagesStatus(ctx context.Context, transactionHash *felt.Felt) ([]MessageStatusResp, error) {
```
Comment (rianhughes) [inline rpc/transaction.go]: transactionHash is an ethereum transaction hash, so should be a common.Hash

File: rpc/types_transaction_receipt.go [hunk 2]
```diff
@@ -28,6 +28,15 @@ type MsgFromL1 struct {
 	Payload []*felt.Felt `json:"payload"`
 }
 
+type MessageStatusResp struct {
+	// The hash of a Starknet transaction
```
Comment (rianhughes) [inline rpc/types_transaction_receipt.go]: nit:The hash of an l1 handler transaction

File: rpc/transaction.go (hunk 1 above)
Comment (thiagodeev) [reply rpc/transaction.go]: Thanks! Done

File: rpc/types_transaction_receipt.go (hunk 2 above)
Comment (thiagodeev) [reply rpc/types_transaction_receipt.go]: Thanks! Done

File: rpc/transaction.go [hunk 3]
```diff
@@ -71,3 +71,20 @@ func (provider *Provider) GetTransactionStatus(ctx context.Context, transactionH
...
+
+// Given an L1 tx hash, returns the associated l1_handler tx hashes and statuses for all L1 -> L2 messages sent by the l1 transaction, ordered by the L1 tx sending order
+//
+// Parameters:
+// - ctx: the context.Context object for cancellation and timeouts.
+// - transactionHash: The hash of the L1 transaction that sent L1->L2 messages
+// Returns:
+// - [] MessageStatusResp: An array containing the status of the messages sent by the L1 transaction
+// - error, if one arose.
+func (provider *Provider) GetMessagesStatus(ctx context.Context, transactionHash NumAsHex) ([]MessageStatusResp, error) {
```
Comment (rianhughes) [inline rpc/transaction.go:83]: Lets add a placeholder test for this as well

**Review by rianhughes — APPROVED** [review]

**Pull request merged**